File: problem-test/src/main/java/com/dudgns/problemtest/entity/carManagement/CarCategoryMappingEntity.java
package com.dudgns.problemtest.entity.carManagement;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "car_category_mapping")
public class CarCategoryMappingEntity {
    @Id
    @Column(name = "car_idx")
    private Long carIdx;
    @Column(name = "category_type")
    private Long categoryType;
    @Column(name = "category_name")
    private String categoryName;
}

File: problem-test/commons/project-dudgns-data-entity/src/main/kotlin/dudgns/com/backend/commons/data/entity/problemTest/CarCategoryMappingEntity.kt
package dudgns.com.backend.commons.data.entity.problemTest

import dudgns.com.backend.commons.data.entity.problemTest.id.CarCategoryMappingId
import jakarta.persistence.Column
import jakarta.persistence.EmbeddedId
import jakarta.persistence.Entity
import jakarta.persistence.Table

@Entity
@Table(name = "car_category_mapping")
class CarCategoryMappingEntity {
    constructor(carCategoryMappingId: CarCategoryMappingId?, categoryName: String?) {
        this.carCategoryMappingId = carCategoryMappingId
        this.categoryName = categoryName!!
    }

    @EmbeddedId
    var carCategoryMappingId: CarCategoryMappingId? = null

    @Column(name = "category_name")
    var categoryName: String = ""
}